### PR TITLE
feat: update is.js and use the new URL mechanics

### PR DESCRIPTION
**Pull request by bobylito**
With this PR, the app will use the new routing (url sync) mechanism. This provides several benefits:
 - cleaner and customizable URLs
 - the layout selection is also in sync

Example of URLs:
 - layout in the URL: https://deploy-preview-52--examiner-trust-47064.netlify.com/?lang=enUS&layout=list
 - lot of filters: https://deploy-preview-52--examiner-trust-47064.netlify.com/?playerClass=Mage&standardOnly=true&cost=7&rarity=Epic&lang=enUS&layout=list

### CODE REVIEW

File: src/js/algolia-instantsearch-conf.js [hunk 1]
```diff
@@ -5,13 +5,68 @@ let instantsearch = require('instantsearch.js');
 import languageSelect from './is-custom/language-select.js';
 instantsearch.widgets.languageSelect = languageSelect;
 
+import layoutSelect from './is-custom/layout-picker.js';
+
+const separator = '_';
+const stateMapping = {
+  stateToRoute(uiState) {
+    return {
+      query: uiState.query,
```
Comment (Haroenv) [inline src/js/algolia-instantsearch-conf.js]: Would maybe nice to destructure eagerly :)

**Comment by Shipow** [comment]
this is awesome <3

**Comment by bobylito** [comment]
Let's merge then?

File: src/js/algolia-instantsearch-conf.js (hunk 1 above)
Comment (bobylito) [reply src/js/algolia-instantsearch-conf.js]: Indeed. I'm gonna try, if it works I'll refactor. If not let's push that to later, ok?

**Pull request merged**